### PR TITLE
Resolves non-gradient bug of SSIM Loss

### LINKED ISSUE #5668: SSIMLoss result does not have gradient
**Describe the bug**
SSIMLoss result does not have gradient
@PedroFerreiradaCosta 
@wyli 

**To Reproduce**
```
import torch
from monai.losses.ssim_loss import SSIMLoss

x = torch.ones([1,1,10,10])/2
y = torch.ones([1,1,10,10])/2
y.requires_grad_(True)
data_range = x.max().unsqueeze(0)
loss = SSIMLoss(spatial_dims=2)(x,y,data_range)
print(loss.requires_grad)
```

**Expected behavior**
`loss.requires_grad` will be False. But it should be True

**Screenshots**

**Environment**

Ensuring you use the relevant python executable, please paste the output of:

```
python -c 'import monai; monai.config.print_debug_info()'
```

**Additional context**

### PR DESCRIPTION
**Pull request by PedroFerreiradaCosta**
Fixes #5668 .

### Description
The SSIM loss wasn't feeding  into autograd - it wasn't allowing gradient to be calculated. Now they can be calculated.
As per my previous PR (#5550), the loss function is now calculated using the SSIMMetric (and not the other way around). The SSIMMetric in inheriting  from class `IterationMetric` that runs `.detach()` on all inputs before running `._compute_tensor()`.
https://github.com/Project-MONAI/MONAI/blob/0737a33d62ce1e18023712a000828235b7758536/monai/metrics/metric.py#L70
I'm now calling  the `SSIMMetric()._compute_tensor()` explicitly, removing the step where it detaches from the current graph.

**To Reproduce**
```
import torch
from monai.losses.ssim_loss import SSIMLoss

x = torch.ones([1,1,10,10])/2
y = torch.ones([1,1,10,10])/2
y.requires_grad_(True)
data_range = x.max().unsqueeze(0)
loss = SSIMLoss(spatial_dims=2)(x,y,data_range)
print(loss.requires_grad)
```

**Expected behavior**
`loss.requires_grad` will now be True.

### Types of changes
- [x] Non-breaking change (fix or new feature that would not break existing functionality).

### CODE REVIEW

**Comment by wyli** [comment]
thanks, could you please add a unit test

**Review by Can-Zhao — APPROVED** [review]
Thank you so much!!

**Comment by wyli** [comment]
merging this for now, please feel free to follow up if you have further code changes.
